source/extensions/filters/udp/udp_proxy/hash_policy_impl.h

```diff
@@ -32,10 +27,27 @@ class HashPolicyImpl : public Udp::HashPolicy {
 
   using HashMethodPtr = std::unique_ptr<HashMethod>;
 
+  HashPolicyImpl(HashMethodPtr hash_impl) : hash_impl_(std::move(hash_impl)) {}
```

Comment:
is this meant to be private to enforce creation through the factory?

Reply:
The factory is a different class so I think I need a public constructor.

Reply:
I might try to embed the factory class.